### ISSUE TITLE
Code for the server?

### ISSUE THREAD

**Issue by c-w** [UniversalDataTool/universal-data-tool#16]
[use-server.js](https://github.com/UniversalDataTool/universal-data-tool/blob/b96b9279e6b6b823249da72433f8daa84c2fc49f/src/utils/file-handlers/use-server.js) makes requests to an API server which is hosted at [udt-collaboration.workaround.now.sh](https://udt-collaboration.workaround.now.sh/README.md). Are there plans to make the code for the server open source so that users can self-host the entire stack for this tool?

**Comment by seveibar** [comment]
Yes! The server uses a simple JSON PATCH system. We just introduced collaboration last week so we're still working out how we want the protocol to work (issues like #12, there are also possible race conditions when samples are deleted/inserted) and making sure we have the right setup to prevent abuse. Because universaldatatool.com deals with more data than most local instances, we'll probably simplify the code and remove some of the external services that the current implementation relies on (i.e. have it run when you run`npm start` without need to run postgres, redis etc.)

**Comment by seveibar** [comment]
Probably a sqlite version.

**Comment by seveibar** [comment]
There's also additional server functionality coming, like file upload!
